homeassistant/components/scene/knx.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for KNX scenes.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/scene.knx/
+"""
+import asyncio
+
+import voluptuous as vol
+
+from homeassistant.components.knx import ATTR_DISCOVER_DEVICES, DATA_KNX
+from homeassistant.components.scene import CONF_PLATFORM, Scene
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+CONF_ADDRESS = 'address'
+CONF_SCENE_NUMBER = 'scene_number'
+
+DEFAULT_NAME = 'KNX SCENE'
+DEPENDENCIES = ['knx']
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'knx',
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_ADDRESS): cv.string,
+    vol.Required(CONF_SCENE_NUMBER): cv.positive_int,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices,
+                         discovery_info=None):
+    """Set up the scenes for KNX platform."""
+    if DATA_KNX not in hass.data \
```

Comment:
This is not possible because you have `knx` as a dependency and so setup never gets called unless knx has been setup correctly.

Also, just returning `False` is weird. Then it will never be setup?